systems/plants/RigidBodyManipulatorURDF.cpp

```diff
@@ -0,0 +1,513 @@
+#include <string>
+#include <fstream>
+#include <sstream>
+
+#include "tinyxml.h"
+#include "RigidBodyManipulator.h"
+#include "joints/drakeJointUtil.h"
+#include "joints/HelicalJoint.h"
+#include "joints/PrismaticJoint.h"
+#include "joints/RevoluteJoint.h"
+#include "joints/QuaternionFloatingJoint.h"
+#include "joints/RollPitchYawFloatingJoint.h"
+
+using namespace std;
+
+// todo: rectify this with findLinkId in the class (which makes more assumptions)
+int findLinkIndex(RigidBodyManipulator* model, string linkname)
+{
+  int index = -1;
+  for (int i = 0; i < model->bodies.size(); i++) {
+    if (linkname.compare(model->bodies[i]->linkname) == 0) {
+      index = i;
+      break;
+    }
+  }
+  return index;
+}
+
+int findLinkIndexByJointName(RigidBodyManipulator* model, string jointname)
+{
+  int index = -1;
+  for (int i = 0; i < model->bodies.size(); i++) {
+    if (model->bodies[i]->hasParent() && jointname.compare(model->bodies[i]->getJoint().getName())==0) {
+      index = i;
+      break;
+    }
+  }
+  return index;
+}
+
+// only writes values if they exist
+bool parseVectorAttribute(TiXmlElement* node, const char* attribute_name, Vector3d &val)
+{
+  const char* attr = node->Attribute(attribute_name);
+  if (attr) {
+    stringstream s(attr);
+    s >> val(0) >> val(1) >> val(2);
+    return true;
+  }
+  return false;
+}
+
+void poseAttributesToTransform(TiXmlElement* node, Matrix4d& T)
+{
+  double x = 0.0, y = 0.0, z = 0.0, roll = 0.0, pitch = 0.0, yaw = 0.0;
+
+  const char* attr = node->Attribute("xyz");
+  if (attr) {
+    stringstream s(attr);
+    s >> x >> y >> z;
+  }
+
+  attr = node->Attribute("rpy");
+  if (attr) {
+    stringstream s(attr);
+    s >> roll >> pitch >> yaw;
+  }
+
+  T << cos(yaw) * cos(pitch), cos(yaw) * sin(pitch) * sin(roll) - sin(yaw) * cos(roll), cos(yaw) * sin(pitch) * cos(roll) + sin(yaw) * sin(roll), x, sin(yaw) * cos(pitch), sin(yaw) * sin(pitch) * sin(roll) + cos(yaw) * cos(roll), sin(yaw) * sin(pitch) * cos(roll) - cos(yaw) * sin(roll), y, -sin(
+      pitch), cos(pitch) * sin(roll), cos(pitch) * cos(roll), z, 0, 0, 0, 1;
```

Comment:
do we have one that takes xyz,rpy -> T?  rpy2rotmat makes the rest of the comma initialization pretty clunky here.